sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2197,13 +2197,15 @@ object functions {
   def round(e: Column): Column = round(e, 0)
 
   /**
-   * Round the value of `e` to `scale` decimal places with HALF_UP round mode
+   * Round the value of `e` to `scale` decimal places with given round mode, default: HALF_UP
    * if `scale` is greater than or equal to 0 or at integral part when `scale` is less than 0.
    *
    * @group math_funcs
    * @since 1.5.0
    */
-  def round(e: Column, scale: Int): Column = withExpr { Round(e.expr, Literal(scale)) }
+  def round(e: Column, scale: Int, mode: String = "half_up"): Column = withExpr {
```

Comment:
This breaks binary compatibility in Java because the default argument doesn't work. we should overload it.

Reply:
Thanks, that fixed it. I'm currently fixing the schema mismatch fails. But i'm not sure why the kubernetes integration test is failing, do you have any input on that, please ?